dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs

```diff
@@ -24,6 +24,16 @@ public sealed class HandlebarsPlannerOptions : PlannerOptions
     /// </remarks>
     public Func<Task<string>>? GetAdditionalPromptContext { get; set; }
 
+    /// <summary>
+    /// Callback that returns an override for the CreatePlan prompt.
+    /// </summary>
+    /// <remarks>
+    /// The callback should return a valid Handlebars template string. If this is set, the planner will use this prompt instead of the default prompt.
+    /// Devs can select any partial defined in Planners.Handlebars.CreatePlanPromptPartials namespace when constructing their own prompt.
+    /// No partials are included by default; make sure to select partials such as "{{> UserGoal }}" or {{> AdditionalContext}} if needed.
+    /// </remarks>
+    public Func<string>? GetCreatePlanPrompt { get; set; }
```

Comment:
Suggest changing the name to represent its a handler that will be used as a callback.

```suggestion
    public Func<string>? CreatePlanPromptHandler { get; set; }
```

Reply:
Done